scripts/archive_links.py

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/env python3
+
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+# ruff: noqa: T201
+
+import argparse
+import json
+import os
+import re
+import sys
+import time
+from typing import Optional
+
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+EXTENSIONS_TO_SCAN = [".py", ".po"]
+FIND_URL_PATTERN = re.compile(r'https?://[^\s<>"]+|www\.[^\s<>"]+')
+IGNORED_DIRECTORIES = ["__pycache__"]
+IGNORE_URL_PATTERNS = [
+    re.compile(r"github\.com"),
+    re.compile(r"wikipedia\.org"),
+    re.compile(r"wikisource\.org"),
+    re.compile(r"namu\.wiki"),
+    re.compile(r"web\.archive\.org"),
+    re.compile(r"archive\.org"),
+    re.compile(r"archive\.is"),
+    re.compile(r"archive\.ph"),
+    re.compile(r"archive\.today"),
+    re.compile(r"docs\.python\.org"),
+    re.compile(r"loc\.gov"),
+    re.compile(r"iso\.org"),
+    re.compile(r"semver\.org"),
+]
+CDX_API_URL = "https://web.archive.org/cdx/search/cdx"
+SAVE_API_URL = "https://web.archive.org/save"
+REQUEST_TIMEOUT = 60
+USER_AGENT = "https://github.com/vacanza/holidays (python)"
+
+
+def find_hyperlinks_in_file(filepath: str) -> list[str]:
+    """
+    Finds all hyperlinks within a single text file, respecting IGNORE_URL_PATTERNS.
+    """
+    try:
+        with open(filepath, encoding="utf-8") as file:
+            file_content = file.read()
+    except Exception as e:
+        print(f"Warning: Could not read file {filepath} due to: {e}", file=sys.stderr)
+        return []
+
+    found_urls = set()
+    try:
+        for match in re.findall(FIND_URL_PATTERN, file_content):
+            if match:
+                ignore = False
+                for ignore_pattern in IGNORE_URL_PATTERNS:
+                    if re.search(ignore_pattern, match):
+                        ignore = True
+                        break
+                if not ignore:
+                    cleaned_match = match.rstrip(")")
+                    found_urls.add(cleaned_match)
+        return sorted(list(found_urls))
```

Comment:
We managed to reduce the initial check time from 7.54373020 to 1.87061540 seconds by switching Python's `any( )` 🎉 
```suggestion
    try:
        found_urls = {
            match.rstrip(")")
            for match in re.findall(FIND_URL_PATTERN, file_content)
            if match and not any(p.search(match) for p in IGNORE_URL_PATTERNS)
        }
        return sorted(found_urls)
```